Created base structure for adding discovery and other algorithms

amit-sharma opened the pull request:
New files added that convey the directory structure.

emrekiciman reviewed (commented):
This is great.  I added a few questions about the structure.

Also: do we want this dataset protocol to integrate well with data handling in other pywhyllm libraries? Or is this only for self-contained pywhy-llm benchmarking for example?

amit-sharma commented on pywhyllm/datasets/dataset.py:
```diff
@@ -0,0 +1,30 @@
+        """
+        pass
+
+    def generate_data(self):
```
right. that's a good idea. In fact, the `generate_data` function could be a standalone utility that can be used for any dataset. So for synthetic datasets, users can call that function first and then create Dataset.

amit-sharma commented:
> This is great. I added a few questions about the structure.
> 
> Also: do we want this dataset protocol to integrate well with data handling in other pywhyllm libraries? Or is this only for self-contained pywhy-llm benchmarking for example?

thanks for the feedback, @emrekiciman . Currently, dowhy expects a pandas dataframe and a Networkx graph (separately). This dataset class is pywhy specific but can be easily used to provide input to dowhy by accessing its members, `graph` and `data`.

Addressed other comments inline.

amit-sharma commented on pywhyllm/datasets/metrics.py:
```diff
@@ -0,0 +1,25 @@
+
+
+def accuracy(edges, true_edges):
```
good point! I also like the second suggestion. I've renamed "datasets" folder to "data".

emrekiciman replied:
thanks!
